cmd/utils/flags.go

```diff
@@ -1414,13 +1421,41 @@ func setTxPool(ctx *cli.Context, cfg *core.TxPoolConfig) {
 	}
 }
 
+func setEthashDatasetDir(ctx *cli.Context, cfg *eth.Config) {
+	switch {
+	case ctx.GlobalIsSet(EthashDatasetDirFlag.Name):
+		cfg.Ethash.DatasetDir = ctx.GlobalString(EthashDatasetDirFlag.Name)
+
+	case (ctx.GlobalBool(ClassicFlag.Name) || ctx.GlobalBool(MordorFlag.Name)) && cfg.Ethash.DatasetDir == eth.DefaultConfig.Ethash.DatasetDir:
+		// ECIP-1099 is set, use etchash dir for DAGs instead
+		home := os.Getenv("HOME")
```

Comment:
This copy-pasta will be able to be replaced with an exported `HomeDir` function, coming from https://github.com/etclabscore/core-geth/pull/191/files#diff-fdd625b1499a66d8f93635113abdd5e0R202.